tsdb/store.go

```diff
@@ -1845,7 +1859,7 @@ func (s *Store) TagValues(auth query.Authorizer, shardIDs []uint64, cond influxq
 	// filter.
 	for _, name := range names {
 		// Determine a list of keys from condition.
-		keySet, err := is.MeasurementTagKeysByExpr(name, cond)
+		keySet, err := is.MeasurementTagKeysByExpr(name, tagKeyExpr)
```

Comment:
This worked before because we *always* had a `_tagKey` field (enforced by the parser). But if you don't have any binary expressions with `_tagKey` but you do have a non-nil expression this returns an empty set.